routing/road_graph.cpp

```diff
@@ -154,14 +166,40 @@ IRoadGraph::RoadInfo::RoadInfo(bool bidirectional, double speedKMPH, initializer
     : m_points(points), m_speedKMPH(speedKMPH), m_bidirectional(bidirectional)
 {}
 
-// IRoadGraph::CrossEdgesLoader ------------------------------------------------
-
-IRoadGraph::CrossEdgesLoader::CrossEdgesLoader(m2::PointD const & cross, TEdgeVector & outgoingEdges)
-    : m_cross(cross), m_outgoingEdges(outgoingEdges)
+// IRoadGraph::CrossOutgoingLoader ---------------------------------------------
+void IRoadGraph::CrossOutgoingLoader::LoadEdge(FeatureID const & featureId,
```

Comment:
This function looks like a twin for CrossIngoing loader. Why not to do following:
```c++
template <typename TFn>
void ForEachEdge(FeatureID const & id, RoadInfo const & info, TFn && fn)
{
  for (size_t i = 0; i < numPoints; ++i)
  {
    m2::PointD const & p = info.m_points[i];
    if (!PointsAlmostEqualAbs(m_cross, p))
      continue;
    if (i != 0)
      fn(info.m_points[i - 1], false /* backward */);
    if (i + 1 < numPoints)
      fn(info.m_points[i + 1], true /* forward */);
  }
}

...

CrossOutgoingLoader::LoadEdges(...)
{
  ForEachEdge([&](m2::PointD const & p, bool forward)
  {
    if (forward || info.m_bidirectional || m_onewayAsBidirectional)
      m_edges.emplace_back(m_cross, p, forward);
  });
}

...
CrossIngoingLoader::LoadEdges(...)
{
  ForEachEdge([&](m2::PointD const & p, bool forward)
  {
    if (forward || info.m_bidirectional || m_onewayAsBidirectional)
      m_edges.emplace_back(p, m_cross, !forward);
  });
}
```